src-graphql/client.js

```diff
@@ -0,0 +1,78 @@
+import GraphQLJSClient from '@shopify/graphql-js-client';
+import types from '../types';
+import base64 from './base64';
+import './isomorphic-fetch';
+
+export default class Client {
+  constructor(config, GraphQLClientClass = GraphQLJSClient) {
+    const apiUrl = `https://${config.domain}/api/graphql`;
+    const authHeader = `Basic ${base64(config.storefrontAccessToken)}`;
+
+    this.graphQLClient = new GraphQLClientClass(types, apiUrl, {
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+        Authorization: authHeader
+      }
+    });
+  }
+
+  fetchAllProducts() {
+    const query = this.graphQLClient.query((root) => {
+      root.add('shop', (shop) => {
+        shop.addConnection('products', {args: {first: 10}}, (products) => {
+          addProductFields(products);
+        });
+      });
+    });
+
+    return this.graphQLClient.send(query).then((response) => {
+      return response.model.shop.products;
+    });
+  }
+
+  fetchProduct(id) {
+    const query = this.graphQLClient.query((root) => {
+      root.add('product', {args: {id: `gid://shopify/Product/${id}`}}, (product) => {
```

Comment:
👍 

Something like `gidFor(model)`, which can grab it's type and id in pretty deterministic ways.